src/qiskit_qasm3_import/expression.py

```diff
@@ -77,10 +97,19 @@ def generic_visit(self, node: ast.QASMNode, context: None = None):
         raise_from_node(node, f"'{node.__class__.__name__}' cannot be resolved into a Qiskit value")
 
     def visit_Identifier(self, node: ast.Identifier):
-        name = node.name
-        if name not in self.symbols:
-            raise_from_node(node, f"name '{name}' is not defined in this scope")
-        symbol = self.symbols[name]
+        cxt = self.context
+        assert isinstance(cxt.symbol_table, SymbolTable)
+        if (symbol := cxt.symbol_table.get(node.name, node)) is not None:
+            return symbol.data, symbol.type
```

Comment:
Nice walrussing - I'm a fan of this sort of use of it.

Reply:
What we really need is [anaphoric macros](https://en.wikipedia.org/wiki/Anaphoric_macro). ( ha ;) )